src/components/Settings/integrations/Mailchimp/MailchimpAccordion.test.tsx

```diff
@@ -149,6 +149,48 @@ describe('MailchimpAccount', () => {
     beforeEach(() => {
       mailchimpAccount = { ...standardMailchimpAccount };
     });
+    it('is connected but with an API key error', async () => {
+      mailchimpAccount.validateKey = false;
+      const mutationSpy = jest.fn();
+      const { queryByText, getByRole } = render(
+        <Components>
+          <GqlMockedProvider<{
+            MailchimpAccount: MailchimpAccountQuery | undefined;
+          }>
+            mocks={{
+              MailchimpAccount: {
+                mailchimpAccount: [mailchimpAccount],
+              },
+            }}
+            onCall={mutationSpy}
+          >
+            <MailchimpAccordion
+              handleAccordionChange={handleAccordionChange}
+              expandedAccordion={IntegrationAccordion.Mailchimp}
+            />
+          </GqlMockedProvider>
+        </Components>,
+      );
+
+      await waitFor(() => {
+        expect(
+          queryByText(
+            'There is an error with your MailChimp connection. Please disconnect and connect to MailChimp again.',
+          ),
+        ).toBeInTheDocument();
+      });
```

Comment:
It's better to use `await findByText` than ` await waitFor`.
```suggestion
      expect(
        await findByText(
          'There is an error with your MailChimp connection. Please disconnect and connect to MailChimp again.',
        ),
      ).toBeInTheDocument();
```